File: craftd-core/src/main/java/com/github/codandotv/craftd/androidcore/data/model/text/TextProperties.kt
package com.github.codandotv.craftd.androidcore.data.model.text

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import com.github.codandotv.craftd.androidcore.data.model.action.ActionProperties
import com.github.codandotv.craftd.androidcore.domain.CraftDAlign
import com.github.codandotv.craftd.androidcore.domain.CraftDTextStyle

@JsonIgnoreProperties(ignoreUnknown = true)
data class TextProperties(
    @JsonProperty("text") val text: String? = null,
    @JsonProperty("textColorHex") val textColorHex: String? = null,
    @JsonProperty("align") val align: CraftDAlign? = null,
    @JsonProperty("textSize") val textSize: String? = null,
    @JsonProperty("backgroundHex") val backgroundHex: String? = null,
    @JsonProperty("textStyle") val textStyle: CraftDTextStyle? = null,
    @JsonProperty("textAllCaps") val textAllCaps: Boolean? = false,
    @JsonProperty("actionProperties") var actionProperties: ActionProperties? = null,
    @JsonProperty("textHtml") val textHtml: String? = null,
)

File: flutter/craftd_widget/lib/data/model/text/text_properties.dart
import '../action/action_properties.dart';

class TextProperties {
  const TextProperties({
    required this.text,
    this.textColorHex,
    this.textSize,
    this.textAllCaps,
    this.backgroundHex,
    this.actionProperties
  });

  final String text;
  final String? textColorHex;
  final String? textSize;
  final bool? textAllCaps;
  final String? backgroundHex;
  final ActionProperties? actionProperties;
}
